pkg/chunk/storage/factory.go

```diff
@@ -37,8 +39,28 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.DurationVar(&cfg.IndexCacheValidity, "store.index-cache-validity", 5*time.Minute, "Period for which entries in the index cache are valid. Should be no higher than -ingester.max-chunk-idle.")
 }
 
-// NewStorageClient makes a storage client based on the configuration.
-func NewStorageClient(cfg Config, schemaCfg chunk.SchemaConfig) (client chunk.StorageClient, err error) {
+// Clients makes the storage clients based on the configuration.
+func Clients(cfg Config, schemaCfg chunk.SchemaConfig) ([]chunk.StorageOpt, error) {
+	opts := []chunk.StorageOpt{}
+	client, err := newStorageClient(cfg, schemaCfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating storage client")
+	}
+
+	opts = append(opts, chunk.StorageOpt{From: model.Time(0), Client: client})
+	if cfg.StorageClient == "gcp" && schemaCfg.BigtableColumnKeyFrom.IsSet() {
+		client, err = gcp.NewStorageClientColumnKey(context.Background(), cfg.GCPStorageConfig, schemaCfg)
```

Comment:
Shouldn't this be a `CachingStorageClient` as well?

Reply:
Good catch! Fixed it.